### PR TITLE
Mapping

### PR DESCRIPTION
**Pull request by mehradans92**
- Added a CG matrix mapper to utils.py

- Added tutorial on generating a CG mapping matrix for a molecule given a bead distribution.

Please run the 08. Molecules CG Mapping.ipynb notebook in examples for review.

### CODE REVIEW

**Comment by whitead** [comment]
Thanks for putting this together @mehradans92! It looks like you accidentally added some unrelated files, could you remove those?

File: htf/utils.py [hunk 1]
```diff
@@ -188,6 +188,29 @@ def find_molecules(system):
 
 # \internal
 # \brief Finds mapping operators for coarse-graining
+def matrix_mapping(molecule, beads_distribution):
+    R""" This will create a M x N mass weighted mapping matrix where M is the number
+        of atoms in the molecule and N is the number of mapping beads.
+    :param molecule: This is atom selection in the molecule.
+    :param beads_distribution: This is a dictionary of beads distribution, where keys should
+     be {Bead_1, Bead_2, ...} and values should be the atoms as strings just like how they appear
```
Comment (whitead) [inline htf/utils.py:196]: Let's just make it a list, as discussed

File: htf/utils.py [hunk 2]
```diff
@@ -188,6 +188,29 @@ def find_molecules(system):
 
 # \internal
 # \brief Finds mapping operators for coarse-graining
+def matrix_mapping(molecule, beads_distribution):
+    R""" This will create a M x N mass weighted mapping matrix where M is the number
+        of atoms in the molecule and N is the number of mapping beads.
+    :param molecule: This is atom selection in the molecule.
```
Comment (whitead) [inline htf/utils.py:194]: Can you explicitly mention this is an MDAnalysis Atoms object?

File: htf/utils.py [hunk 3]
```diff
@@ -188,6 +188,29 @@ def find_molecules(system):
...
+
+    :return: An array of M x N.
+    """
+    Mws_dict = dict(zip(molecule.names, molecule.masses))
+    M, N = len(beads_distribution), len(molecule)
+    CG_matrix = np.zeros((M, N))
+    index = 0
+    for s in range(M):
+        for i, atom in enumerate(beads_distribution['Bead_{}'.format(str(s+1))]):
+            CG_matrix[s, i+index] = [v for k,
```
Comment (whitead) [inline htf/utils.py:207]: What happens if multiple atoms have the same name?

Comment (mehradans92) [reply htf/utils.py:207]: Should still work. Please check the tutorial I added, there is repetition in the atom names.

File: examples/08.  Molecules CG Mapping.ipynb [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+      "  0.         0.         0.         0.         0.         0.\n",
+      "  0.         0.         0.         0.         0.         0.\n",
+      "  0.         0.         0.         0.         0.27291648 0.36354176\n",
+      "  0.36354176]]\n"
+     ]
+    }
+   ],
+   "source": [
+    "# Loading inputs\n",
+    "TPR = '/scratch/awhite38_lab/ff/OPLS/pure_water/non_bonded_traj/nvt_prod_outforce.tpr'\n",
```
Comment (whitead) [inline examples/08.  Molecules CG Mapping.ipynb:103]: We need to use files that are part of the package so others can run it.

**Pull request merged**